WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -571,4 +571,79 @@
       end
     end
   end
+
+  describe 'GET #my_competitions' do
```

Comment:
This is a good and ambitious test! 

Some improvements/suggestions:

* I don't see any test of the ` || Date.today + 20.year` part. If you remove that, I assume the test still passes? You probably need a competition without start date.

* The value of @competitions is not tested. If it's used, it should be tested. If not, it should lose the @.

* The "one assertion per test" pattern gets very wordy and a bit confusing here. I would do them like this:

    context 'when signed in as an organizer user' do
      it 'shows my past and upcoming competitions' do
        sign_in organizer1

        get :my_competitions

        expect(assigns(:not_past_competitions)).to eq [future_competition1, future_competition2, future_competition3]
        expect(assigns(:past_competitions)).to eq [past_competition1, past_competition3]
      end
    end